lib/broccoli/ember-app.js

```diff
@@ -1845,6 +1845,30 @@ function addOutputFile(container, assetPath, options) {
     container[outputFile] = [];
   }
 
+  if (~container[outputFile].indexOf(assetPath)) {
+    var message = [
+      'You have specified a duplicate asset via `app.import(' + assetPath + ')`.',
+      'This can result in very strange behavior in your application as the',
+      'asset historically was included twice. Multiple versions of the same asset',
+      'are also clobbered by the last version added to the Broccoli tree, meaning',
+      'that it only ever included the version of the asset which was last added',
+      'to the Broccoli tree. This should be treated as a bug to address. This',
+      'typically occurs when an asset is imported by a nested addon. Ember CLI',
+      'no longer includes multiple copies by default and instead is using only',
+      'the _last_ occurrence imported.'
+    ];
+    this.project.ui.writeWarnLine(message.join(' '));
```

Comment:
I don't think I'm a fan of warning here. We should basically just do the right thing. 

Warning here will be super noisy for folks, and due to our config + nested addon story there is very little they can actually do about it (other than possibly storing some flag on the project and checking for it, which would only work for newer versions of that addon).